File: Python/Stack.py
from inspect import stack

# Create a stack
def create_stack():
    stack = [];
    return stack;

# State whether stack is empty or not;
def stack_empty(stack):
    return len(stack) == 0;

# Push an item into the stack
def push(stack, item):
    stack.append(item);
    print("Pushed item: " + item + " to stack");

# Pop an element from the stack
def pop(stack):
    item = stack[len(stack)-1];
    stack.pop();
    print("Popped item: " + item + " from stack");

# Show the stack in terminal
def show_stack(stack):
    print(stack);

def peek_stack(stack):
    print(stack[len(stack)-1]);

stack = create_stack();
push(stack, str(1));
push(stack, str(2));
push(stack, str(3));
push(stack, str(4));

peek_stack(stack);
show_stack(stack);
    
    

File: C++/Stack.cpp
// Implementing a Stack Data Structure using C++

#include <iostream>

using namespace std;

class Stack
{
  int top;
	public;
	int a[10]; //
	Stack() {
		top = -1;
	}

	void push(int x);
	int pop();
	void isEmpty();
	void isFull();
};

// Function to push data into stack
void Stack::push(int x) {
	if (top >= 10) {
		cout << "Stack Overflow \n"
	}
	else 
	{
		a[++top] = x;
		cout << "Element Inserted \n"
	}
};

void Stack::pop() {
	if (top < 0) {
		cout << "Stack Underflow \n"
			return 0;
	}
	else 
	{
		int d = a[top--];
		return d;
	}
};

void Stack::isEmpty() {
	if (top < 0) 
	{
		cout << "Stack is empty \n";
	}
	else
	{
		cout << "Stack is not empty \n";
	}
}

void Stack::isFull() {
	if (top > 10) {
		cout << "Stack is full \n";
	} 
	else
	{
		cout << "Stack is not full \n"
	}
}

int main() {
	
	Stack S1;
	S1.push(99);
	S1.push(999);
	S1.push(9999);

	S1.push(10000);
	S1.pop();

	S1.isEmpty();
	}
};

